### PR TITLE
Ap/async logger

### PR DESCRIPTION
**Pull request by aldopareja**

### CODE REVIEW

**Review by RobotSail — APPROVED** [review]
LGTM

**Comment by russellb** [comment]
@RobotSail @aldopareja, we really, REALLY can't merge PRs that fail the DCO check in CI. This is a legal requirement for how we accept contributions to the repos.

File: requirements.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ torch>=2.3.0a0
 transformers>=4.41.2
 datasets>=2.15.0
 numba
-numpy
+numpy==1.26.4
```
Comment (russellb) [inline requirements.txt:10]: specific version pinning like this is very problematic as this gets included in a larger distribution. This should be removed ASAP

**Comment by RobotSail** [comment]
@russellb You are right about the DCO and version pinning, thank you for pointing it out.

**Pull request merged**